analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ExplodedNode.cs

```diff
@@ -69,7 +69,7 @@ public int AddVisit()
         && other.State.Equals(State);
 
     public override string ToString() =>
-        Operation is null
-            ? $"Block #{Block.Ordinal}, Branching{Environment.NewLine}{State}"
-            : $"Block #{Block.Ordinal}, Operation #{index}, {Operation.Instance.Serialize()}{Environment.NewLine}{State}";
+        Operation.Instance is { } operation
+            ? $"Block #{Block.Ordinal}, Operation #{index}, {operation.Serialize()}{Environment.NewLine}{State}"
```

Comment:
[education] can this have an impact on the Security FE ?

Reply:
Yes. The FE frontend might be affected by the change from `class` to `readonöly struct`. `IOperationWrapperSonar` could be `null` before but is never `null` now (instead the wrapped `IOperationWrapperSonar.Instance` can be `null` now).

Reply:
can you check with them on slack if they have a ticket in JIRA for updating the CFG library (they should), and after this gets merged, to add a note about this expected change?

Reply:
from what I understand, it will impact in the sense that when doing the update they may need to change some of the code.

Could it lead to unexpected failures if some test cases are missing from the front-end? 
Can you think of an example of practical code snippet where the behavior would change?

Reply:
And also they will need to update all the property names, I guess

Reply:
> Can you think of an example of practical code snippet where the behavior would change?

I don't know if the FE access the `IOperationWrapperSonar` at all. It is in most parts used internally in the SE-engine. The biggest impact is the `null` behavior which results in changes likes that one in this PR:
```diff
- var successors = current.Operation == null ? ProcessBranching(current) : ProcessOperation(current);
+ var successors = current.Operation.Instance == null ? ProcessBranching(current) : ProcessOperation(current);
```
[here](https://github.com/SonarSource/sonar-dotnet/pull/8105/files#diff-53bfeeb60e2c0755024a073063996ebb8b20e82bd6af9ee79bb5acd416e46fe6L81-R81)

This is a very subtle change and can only be caught by unit tests. `current.Operation == null` is never true, but the compiler is not complaining here.

> And also they will need to update all the property names, I guess

No. The public surface stays the same. It is the change from `class` to `struct` that is problematic (but needed).

Reply:
Ok, this is worthy to mention in a JIRA ticket to keep in mind.